clang-tools-extra/docs/clang-tidy/checks/bugprone/unsafe-crtp.rst

```diff
@@ -0,0 +1,62 @@
+.. title:: clang-tidy - bugprone-unsafe-crtp
+
+bugprone-unsafe-crtp
+====================
+
+Finds CRTP used in an error-prone way.
+
+If the constructor of a class intended to be used in a CRTP is public, then
+it allows users to construct that class on its own.
```

Comment:
Just by reading the documentation, this single sentence seems incomplete. It does not look like it could convince someone without intricate C++ knowledge about why allowing such thing is bad. No need for something elaborate here, but perhaps another sentence that summarises the _"Why?"_ should fit here.

Reply:
Quoting [cppreference](https://en.cppreference.com/w/cpp/language/crtp):

> The [Curiously Recurring Template Pattern](https://en.wikipedia.org/wiki/Curiously_Recurring_Template_Pattern) is an idiom in which a class X derives from a class template Y, taking a template parameter Z, where Y is instantiated with Z = X.

What these lines in the documentain say is bad because they break the `Y is instantiated with Z = X` part. It doesn't do what this idiom says it should do, otherwise from the language's point of view it's perfectly legal.